lib/samples/download_preplanned_map_area/download_preplanned_map_area_sample.dart

```diff
@@ -34,130 +32,269 @@ class DownloadPreplannedMapAreaSample extends StatefulWidget {
+      trailing: const Icon(Icons.download),
+      onTap: () => downloadOfflineMap(mapArea),
+    );
+  }
 
-    final downloadParams = await offlineMapTask
+  // Download an offline map for a provided preplanned map area.
+  void downloadOfflineMap(PreplannedMapArea mapArea) async {
+    // Create default parameters using the map area.
+    final defaultDownloadParams = await _offlineMapTask
         .createDefaultDownloadPreplannedOfflineMapParameters(
-            preplannedMapArea: selectedPreplannedArea);
-
-    // This sample map is not setup for updates
-    downloadParams.updateMode = PreplannedUpdateMode.noUpdates;
+            preplannedMapArea: mapArea);
 
-    final mapDownloadDir = await _getMapDownloadDirectory();
-    final downloadMapJob =
-        offlineMapTask.downloadPreplannedOfflineMapWithParameters(
-            parameters: downloadParams,
-            downloadDirectoryUri: mapDownloadDir.uri);
+    // Set the required update mode. This sample map is not setup for updates so we use noUpdates.
+    defaultDownloadParams.updateMode = PreplannedUpdateMode.noUpdates;
 
-    final downloadResult = await downloadMapJob.run();
+    // Create a directory for the map in the downloads directory.
+    final mapDir =
+        Directory('${_downloadDirectory.path}/${mapArea.portalItem.title}');
+    mapDir.createSync();
 
-    // Return downloaded map
-    if (!downloadResult.hasErrors) {
-      return downloadResult.offlineMap;
-    }
+    // Create and run a job to download the offline map using the default params and download path.
+    final downloadMapJob =
+        _offlineMapTask.downloadPreplannedOfflineMapWithParameters(
+            parameters: defaultDownloadParams,
+            downloadDirectoryUri: mapDir.uri);
 
-    return null;
+    // Associate the job with the map area and update the UI.
+    setState(() => _preplannedMapAreas[mapArea] = downloadMapJob);
+    // Update the UI when the progress changes.
+    downloadMapJob.onProgressChanged.listen((_) => setState(() {}));
+    downloadMapJob.run();
   }
 
-  Future<Directory> _getMapDownloadDirectory() async {
+  // Create the directory for downloading offline map areas into.
+  Future<Directory> createDownloadDirectory() async {
     final documentDir = await getApplicationDocumentsDirectory();
-    final mapDir = Directory('${documentDir.path}/offline_map');
-    if (mapDir.existsSync()) {
-      mapDir.deleteSync(recursive: true);
+    final downloadDir = Directory('${documentDir.path}/preplanned_map_sample');
+    if (downloadDir.existsSync()) {
+      downloadDir.deleteSync(recursive: true);
     }
+    downloadDir.createSync();
+    return downloadDir;
+  }
 
-    mapDir.createSync();
-    return mapDir;
+  // Sets the provided map to the map view and updates the viewpoint.
+  void setMapAndViewpoint(ArcGISMap map) {
+    // Set the map to the map view and update the UI to reflect the newly selected map.
+    setState(() => _mapViewController.arcGISMap = map);
```

Comment:
I'm not sure if we should put the map assignment inside the `setState` callback. They're not really related.

That would make this another `setState(() {})`. I generally don't like that, but maybe that's fine for a sample. The alternative would be something like defining a StateWidget that represents an individual item in the ListView, but that might add a bunch of complexity that doesn't help the tutorial.